### PR TITLE
Thomas/display decision rule exec ast

### PR DESCRIPTION
**Pull request by balzdur**
In this PR : add rule AST on decision
- add ast on decision rule execution
- improve the decision repository
- rework the `RuleDetails` component to display rule AST

### CODE REVIEW

File: packages/app-builder/src/components/Decisions/RulesDetail.tsx [hunk 1]
```diff
@@ -1,116 +1,152 @@
...
+                    {rule.formula ? (
+                      <RuleFormula
+                        formula={rule.formula}
+                        databaseAccessors={databaseAccessors}
+                        payloadAccessors={payloadAccessors}
+                        astOperators={astOperators}
+                        dataModel={dataModel}
+                        customLists={customLists}
+                        triggerObjectType={triggerObjectType}
+                      />
```
Comment (balzdur) [inline packages/app-builder/src/components/Decisions/RulesDetail.tsx:105]: This is the key part of this PR: add `<RuleFormula />` for each rule exec on a decision

File: packages/app-builder/src/components/Decisions/RulesDetail.tsx [hunk 2]
```diff
@@ -1,116 +1,152 @@
...
+    backendAst: formula,
+    localValidation: null,
+    databaseAccessors,
+    payloadAccessors,
+    astOperators,
+    dataModel,
+    customLists,
+    triggerObjectType,
+    onValidate: () => {},
+  });
```
Comment (balzdur) [inline packages/app-builder/src/components/Decisions/RulesDetail.tsx:145]: This still feels wrong to use a "edit" specific hook in order to display the AST (ex: we need to feed `onValidate: () => {}`)

There might be a better abstraction to handle the whole thing but :
- this work like this (so let's move on)
- I prefer to wait for the next steps in order to find a better abstraction (= inject intermediate evaluation errors)

Comment (Pascal-Delange) [reply packages/app-builder/src/components/Decisions/RulesDetail.tsx:145]: yes totally agreed

**Review by Pascal-Delange — APPROVED** [review]
👍  nice

File: packages/app-builder/src/models/decision.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+    id: dto.id,
+    createdAt: dto.created_at,
+    triggerObject: dto.trigger_object,
+    triggerObjectType: dto.trigger_object_type,
+    outcome: dto.outcome,
+    scenario: {
+      id: dto.scenario.id,
+      name: dto.scenario.name,
+      description: dto.scenario.description,
+      version: dto.scenario.version,
```
[the remark this answers is not in the document]
Comment (balzdur) [reply packages/app-builder/src/models/decision.ts:63]: I'll do it in a separate PR, and consume it in the UI to add links

**Pull request merged**